### ISSUE TITLE
BB: extrude a line, then solidify. 

### ISSUE THREAD

**Issue by theoryshaw** [IfcOpenShell/IfcOpenShell#2077]
Is there an IFC construct that emulates the following?...

![2022-03-03_15-03-49_Blender_ DDropboxGitLab101_W_33rd_StModels_a_blender](https://user-images.githubusercontent.com/507113/156652943-213df331-2a67-42ee-b89f-4d0ef2698872.gif)

If so, would it be possible to roundtrip this into BB, somehow?  That is, recreate these modifiers upon import?

**Comment by Moult** [comment]
Yes the construct you're looking for is an axis and I've already built the foundation for it, but now a UI needs to be built to allow parametric editing of it, a bit similar to how editing layered slabs already auto applies the solidify modifier.

**Comment by Moult** [comment]
BTW is there something lacking in the wall tools (extend / join / mitre / etc) that don't already provide similar functionality?

**Comment by theoryshaw** [comment]
it's not necessarily associated with the walls, but things like 'flashing' where it makes it easier to edit them when you only have to edit a line. 

file example: https://gitlab.com/openingdesign/101_W_33rd_St/-/blob/492b8ef143cb909a295d20d434f8076a086800ac/Models%20and%20CAD/IFC/101_W_33rd_St_Details_A.ifc
 
![2022-04-30_23-00-57_Blender_blender (2)](https://user-images.githubusercontent.com/507113/166131669-1c0e6aee-dee4-40ef-83ee-7aee18652d60.gif)

**Comment by Moult** [comment]
Understood, so the correct solution would be to provide an easy line-based UI to edit the axis for layer set usages or profile set usages.

**Comment by theoryshaw** [comment]
Yep, that would work!

**Comment by theoryshaw** [comment]
Implementing this would be boss!
https://github.com/hlorus/CAD_Sketcher/issues/26

**Comment by Moult** [comment]
![Peek 2023-03-03 18-05](https://user-images.githubusercontent.com/88302/222653842-7764d998-c50c-4b1f-889e-565d07845ab7.gif)

**Comment by theoryshaw** [comment]
@Moult this is awesome functionality!.. but I was actually referring to something like this instead...

![2023-03-03_10-36-28_Blender_blender](https://user-images.githubusercontent.com/507113/222776341-a8fc9a46-59c2-496a-8d48-b5dbb1773375.gif)

**Comment by Moult** [comment]
Well you can almost basically do this already, just with a few more clicks.

1. In Scene Properties > IFC Geometry > IFC Profiles add a new arbitrary profile.
2. Edit that profile.
3. Draw out your shape. With the CAD Tool, Shift-O offsets it, and use "F" to join the edges.
4. Launch types manager.
5. Create a new covering type for your flashing as an extruded profile in the types manager.

Is that sufficient? This way you can properly organise your profiles and reuse them.

**Comment by theoryshaw** [comment]
seems like [IfcCenterLineProfileDef](https://ifc43-docs.standards.buildingsmart.org/IFC/RELEASE/IFC4x3/HTML/lexical/IfcCenterLineProfileDef.htm) might be a good fit here.
